gk/main.c

```diff
@@ -246,9 +246,26 @@ extract_packet_info(struct rte_mbuf *pkt, struct ipacket *packet)
 
 	default:
 		packet->flow.proto = 0;
-		RTE_LOG(NOTICE, GATEKEEPER,
-			"gk: unknown network layer protocol %" PRIu16 "!\n",
-			ether_type);
+
+		/*
+		 * If this field is >= 0x0600, it is an EtherType
+		 * field from the Ethernet II standard.
+		 *
+		 * If this field is <= 0x05DC, it is a length
+		 * field from the 802.3 standard. Any other
+		 * value is invalid. We only log this when in
+		 * debug mode.
+		 */
+		if (ether_type < 0x0600) {
```

Comment:
Doesn't DPDK already have this constant? If so, we could use it here instead of the magic number. If not, shouldn't we make this constant available in Gatekeeper as a define? The same comment applies to file `gt/main.c` in this patch.